cuml/forest_inference_demo.ipynb

```diff
@@ -0,0 +1,341 @@
+    "train_size = 0.8\n",
+    "\n",
+    "#split the dataset into training and validation splits\n",
+    "X_train, X_validation, y_train, y_validation = train_test_split(\n",
+    "    X, y, train_size=train_size)\n",
+    "print(\"Shape of the training data : \", np.shape(X_train))\n",
+    "print(\"Shape of the validation data : \", np.shape(X_validation))\n",
+    "if select_model == 'xgboost':\n",
+    "    trained_model_preds, model_path = train_xgboost_model(X_train, y_train,\n",
+    "                                                          X_validation,\n",
+    "                                                          y_validation,\n",
+    "                                                          num_rounds,\n",
+    "                                                          classification)\n",
+    "elif select_model == 'lightgbm':\n",
+    "    trained_model_preds, model_path = train_lightgbm_model(X_train,\n",
+    "                                                           y_train,\n",
+    "                                                           X_validation,\n",
+    "                                                           num_rounds)\n",
+    "else:\n",
+    "    raise(\" This model is not supported, please choose either\"\n",
+    "          \" an xgboost model or lightgbm model\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### The load function of the ForestInference class accepts the following parameters:\n",
+    "        filename : str\n",
+    "           Path to saved model file in a treelite-compatible format\n",
+    "           (See https://treelite.readthedocs.io/en/latest/treelite-api.html\n",
+    "        output_class : bool\n",
+    "           If true, return a 1 or 0 depending on whether the raw prediction\n",
+    "           exceeds the threshold. If False, just return the raw prediction.\n",
+    "        threshold : float\n",
+    "           Cutoff value above which a prediction is set to 1.0\n",
+    "           Only used if the model is classification and output_class is True\n",
+    "        algo : string name of the algo from (from algo_t enum)\n",
+    "             'NAIVE' - simple inference using shared memory\n",
+    "             'TREE_REORG' - similar to naive but trees rearranged to be more\n",
+    "                              coalescing-friendly\n",
+    "             'BATCH_TREE_REORG' - similar to TREE_REORG but predicting\n",
+    "                                    multiple rows per thread block\n",
+    "        model_type : str\n",
+    "            Format of saved treelite model to load.\n",
+    "            Can be 'xgboost', 'lightgbm', or 'protobuf'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# load the saved xgboost model to return the forest in the format used as an input by the forest inference library\n",
```

Comment:
I think this last cell is basically 100% of the demo... we should probably split it into two notebooks or otherwise simplify. I think the message of how to use FIL gets lost with all of the data setup, helper functions, and model saving. I'd suggest doing the following:

* Switch to just XGBoost (we can add a comment to say lightgbm works the same way and we can later add another notebook for that.
* Only do classification - that will eliminate the need for the simulate_data helper function and a lot of other helpers.
 * Let's generally comb throw and toss out any unnecessary code.